### PR TITLE
Renew feature

### PR DESCRIPTION
**Pull request by chang111**
To fix the issue [3412](https://github.com/pymc-devs/pymc3/issues/3412)
please review.

### CODE REVIEW

**Comment by chang111** [comment]
@twiecki 
I have open a new request.
please review it.

File: pymc3/diagnostics.py [hunk 1]
```diff
@@ -5,6 +5,8 @@
 from .util import get_default_varnames
 from .backends.base import MultiTrace
 
+from plots.__init__ import map_args
```
Comment (twiecki) [inline pymc3/diagnostics.py]: Does `from .plots import map_args` also work?

**Comment by twiecki** [comment]
@chang111 There are some test failures which tell you what's wrong. You can them locally using pytest.

**Comment by junpenglao** [comment]
That's not quite how you should do it. The map_arg is because we are now importing function from another library and mapping the input. But in this case because the function `pm.summary` is still in the code base, it is easier just to pop the kwarg and output a warning. See eg (deprecation of `njobs` kwarg):

https://github.com/pymc-devs/pymc3/blob/master/pymc3/sampling.py#L343-L347

**Comment by twiecki** [comment]
@junpenglao is right.

**Comment by chang111** [comment]
@junpenglao Thank you very much for reminding

**Comment by twiecki** [comment]
Seems like tests are still failing. Let me know if you need help with running them.

**Comment by chang111** [comment]
My local environment seems to be slightly different from the online one. Let me adjust it again.

**Comment by chang111** [comment]
The test system does not recognize warnings.

**Comment by chang111** [comment]
Although the travis checks failed, but it does not appear related to this PR.
Tests failed on  import warning. Is there a problem with the setup of the test environment?

File: pymc3/diagnostics.py [hunk 2]
```diff
@@ -97,7 +98,8 @@ def geweke(x, first=.1, last=.5, intervals=20):
         return np.array(zscores)
 
 
-def gelman_rubin(mtrace, varnames=None, include_transformed=False):
+
+def gelman_rubin(mtrace, var_names=None, include_transformed=False,**kwargs):
```
Comment (junpenglao) [inline pymc3/diagnostics.py]:
```suggestion
def gelman_rubin(mtrace, var_names=None, include_transformed=False, **kwargs):
```

File: pymc3/diagnostics.py [hunk 3]
```diff
@@ -163,20 +170,20 @@ def rscore(x, num_samples):
...
+    for var in var_names:
         x = np.array(mtrace.get_values(var, combine=False))
         num_samples = x.shape[1]
         Rhat[var] = rscore(x, num_samples)
 
     return Rhat
 
 
-def effective_n(mtrace, varnames=None, include_transformed=False):
+def effective_n(mtrace, var_names=None, include_transformed=False,**kwargs):
```
Comment (junpenglao) [inline pymc3/diagnostics.py]:
```suggestion
def effective_n(mtrace, var_names=None, include_transformed=False, **kwargs):
```

**Review by junpenglao — APPROVED** [review]

**Comment by chang111** [comment]
@twiecki All check has passed.Please review it.

**Comment by junpenglao** [comment]
Thanks!

**Pull request merged**